src/cli/commands/install.js

```diff
@@ -69,6 +76,43 @@ type Flags = {
   tilde: boolean,
 };
 
+/**
+ * Try and detect the installation method for Yarn and provide a command to update it with.
+ */
+
+function getUpdateCommand(): ?string {
+  // Tarball install
```

Comment:
Instead of doing this inferrence, let's save a token somewhere as part of the packaging process. @cpojer said: 
> Can we somehow add a token to every build indicating the install method? I think that's way better than any sort of inference we might do.

My idea for that was to add an extra `installationMethod` field to `package.json` when building the packages. For example, something like this would be bundled in the Debian package:
```
{
  "name": "yarn",
  "version": "0.16.2",
  "installationMethod": "deb",
```

What do you think? #942 is the issue that covers(-ish) that. If you agree that that's a good idea, let's complete #942 before this one, and then rebase this on top of that.

The main issue I see with inference is when the user has multiple installations of Yarn. That's a bit silly, but they might have installed via the tarball and then also done `npm install -g yarn` for some reason. The instructions might tell them to update the wrong one.